webapi/src/main/java/org/sakaiproject/webapi/controllers/AnnouncementsController.java

```diff
@@ -95,12 +106,18 @@ public List<AnnouncementRestBean> getUserAnnouncements(@PathVariable String user
 	@GetMapping(value = "/sites/{siteId}/announcements", produces = MediaType.APPLICATION_JSON_VALUE)
     public List<AnnouncementRestBean> getSiteAnnouncements(@PathVariable String siteId) throws UserNotDefinedException {
 
+        int numberOfAnnouncements = DEFAULT_NUM_ANNOUNCEMENTS;
+        int numberOfDaysInThePast = DEFAULT_DAYS_IN_PAST;
+
+        //get the Sakai Time for the given java Date
+        Time t = timeService.newTime(getTimeForDaysInPast(numberOfDaysInThePast).getTime());
+
 		Session session = checkSakaiSession();
 
         try {
             Site site = siteService.getSite(siteId);
             String channelRef = announcementService.channelReference(siteId, "main");
-            return ((List<AnnouncementMessage>) announcementService.getMessages(channelRef, null, false, true))
+            return ((List<AnnouncementMessage>) announcementService.getMessages(channelRef, new ViewableFilter(null, t, numberOfAnnouncements, announcementService), false, true))
```

Comment:
my instinct would be to modify the announcementService method getMessages to return a sane, filtered list back ....

Reply:
see my previous comment

Reply:
I copied the logic from AnnoucnementEntityProvider.

So are you saying instead of the logic I added to webapi, just continue sending a null filter to announcementService and if the filter is null, _then_ create a filter there?  based on defaults in announcementService?

what if in the future, you wanted to add options to allow the user to set the number of announcements in the dashboard widget?  (some of the other dashboard widgets have options)... that seems like a much bigger issue than this bug?

Reply:
okay if you think the logic for selecting the number of announcements belongs in dashboard, then your patch makes sense. it does scare me when i see places that need to implement a complex piece of filtering code when the default should be sane and filtered in the first place.

Reply:
I took a closer look at the "Discussions" dashboard widget because it has an "options" checkbox.  but it doesn't seem to save the option state, perhaps is just for filtering option during _that_ page view?  If that's the intent on how the dashboard widgets work (JS only that doesn't save state), I see your point about moving the logic to announcementService.  I'll take a look at that and submit a different PR if I can get it working that way.

any suggestions on what a "sane" filter would be?  10 announcements / 10 days?  (instead of 3 for 10 days).  something else?

Reply:
i would just take the default used from the Announcements synoptic

but again, these changes aren't necessary if it's beyond the scope of what you want to get done here.

Reply:
I just realized that the dashboard tool does paging, so there's no need for adding defaults to the filter. Also, if you add the dashboard to your  "Home" site, the bug doesn't happen there.  It only seems to happen in "course" sites.  I'll need to research that more.  In the meantime, I'll just close this PR, and reopen a new one once I have a better patch in place